src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/Maintainability/DoNotIgnoreMethodResults.cs

```diff
@@ -140,6 +150,13 @@ public override void Initialize(AnalysisContext analysisContext)
                 INamedTypeSymbol? xunitAssertType = compilationContext.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.XunitAssert);
                 INamedTypeSymbol? linqEnumerableType = compilationContext.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemLinqEnumerable);
 
+                if (!(compilationContext.Compilation.SyntaxTrees.FirstOrDefault() is SyntaxTree tree))
+                {
+                    return;
+                }
+
+                var userDefinedMethods = compilationContext.Options.GetAdditionalUseResultsMethodsOption(UserDefinedMethodRule, tree, compilationContext.Compilation, compilationContext.CancellationToken);
```

Comment:
I suggest moving this inside the `RegisterOperationAction` callback. Our options infrastructure ensures that we compute the option value only once per distinct editorconfig value set - i.e. it will hit the cache for all syntax trees that have the same editorconfig that applies to them.